src/lib/reasoners/sat_solver_sig.mli

```diff
@@ -128,17 +134,38 @@ module type S = sig
   (** [get_unknown_reason t] returns the reason Alt-Ergo raised
       [I_dont_know] if it did. If it did not, returns [None]. *)
 
-  val get_value : t -> Expr.t -> Expr.t option
-  (** [get_value t e] returns the value of [e] as a constant expression
-      in the current model generated. Returns [None] if can't decide. *)
-
   val get_objectives : t -> Objective.Model.t option
   (** [get_objectives t] returns the current objective values.
 
       @return [None] if there is no objective or the environment is
       unsatisfiable. *)
 end
 
+type lbool = True | False | Unknown
+
+val pp_lbool : lbool Fmt.t
+
+module type S = sig
+  include Internal
+
+  val get_value : t -> Expr.t list -> Expr.t list option
+  (** [get_value env l] returns the model values of the expressions of [l]
+      in the current generated model of [env].
+
+      @return [None] if the model generation is not enabled or the
+              environment is already unsatisfiable before calling this function.
+      @raise Unsat if the solver found a contradiction. *)
```

Comment:
The documentation is a bit unclear. This function returns `None` if there is no model generated because we don't enable the model generation of the environment `env` was already unsatisfiable before calling `get_value`. W

We raise `Unsat` if we found a contradiction while asserting the model in `get_value`.

We don't display the same message in `Solving_loop`:
```ocaml 
  let handle_get_value loc ~get_value l =
    let l =
      List.map (D_cnf.mk_expr ~loc ~toplevel:false
                  ~decl_kind:Daxiom) l
    in
    match get_value l with
    | Some values ->
      Printer.print_std
        "(@[<v 0>%a@])@,"
        Fmt.(iter ~sep:cut Lists.iter_pair
               ((pair ~sep:sp Expr.pp_smtlib Expr.pp_smtlib) |> parens))
        (l, values)
    | None ->
      recoverable_error "No model produced, cannot execute get-value."
    | exception Sat_solver_sig.Unsat _ ->
      recoverable_error "The model is wrong, cannot execute get-value."
  in
```

So if you got `None`, it means you don't use the API correctly (or the SMT-LIB language), or there is something wrong in our model generation.
If you got `Unsat`, it means there is something wrong in the model.

Reply:
>  If you got Unsat, it means there is something wrong in the model.

I think we should use a different exception here — the `Unsat` exception strongly implies that the *problem* is unsatisfiable, but since this comes from a specific model of the problem, there may be another model that is satisfiable. It is not a problem for `Solving_loop`, but if we ever use `get_value` in another context we take the risk of introducing unsoundness.